File: app/src/main/java/com/emiryanvl/tapreader/domain/usecases/UpdateBookUseCase.java
package com.emiryanvl.tapreader.domain.usecases;

import com.emiryanvl.tapreader.domain.model.Book;
import com.emiryanvl.tapreader.domain.repositories.BookRepository;

import javax.inject.Inject;

public class UpdateBookUseCase {

    private final BookRepository repository;

    @Inject
    public UpdateBookUseCase(BookRepository repository) {
        this.repository = repository;
    }

    public void execute(int id, Book book) {
        repository.updateBook(id, book);
    }
}

File: app/src/main/java/com/emiryanvl/tapreader/domain/usecases/UpdateBookUseCase.kt
package com.emiryanvl.tapreader.domain.usecases

import com.emiryanvl.tapreader.domain.models.Book
import com.emiryanvl.tapreader.domain.repositories.BookRepository
import javax.inject.Inject

class UpdateBookUseCase @Inject constructor(private val repository: BookRepository) {
    suspend operator fun invoke(id: Int, book: Book) = repository.updateBook(id, book)
}